docs/develop/01_sdk/02_cookbook/06_upgrading_to_v0_29/01_backwards_compatibility.md

```diff
@@ -3,52 +3,84 @@ id: v29-backwards-compatibility
-In the verifier flow the `submit-credential` message is affected.
-Both these changes relate to the Credentials API and will require a new version of it released.
-The current specification defines the way for the extension to announce the version of specification it adheres to.
-This makes it possible for the DApp to detect the version mismatch.
-Unfortunately, the extension cannot detect the version mismatch.
-The new specification version should also require the DApp to announce the version of specification it adheres to.
-(This will be a non-enumerable property: `Object.defineProperty(window.kilt, 'meta', { value: { versions: { credentials: '3.0' } }, enumerable: false })`)
+To align with breaking changes to data structures in messaging, credentials, and CTypes, we published a version 3.0 of the [Credentials API specification](https://github.com/KILTprotocol/spec-ext-credential-api) that specifices how browser extensions like the [Sporran credential wallet](https://github.com/BTE-Trusted-Entity/sporran-extension) interact with web applications that produce or consume credentials.
 
-With the versions known the apps might choose to either bail on version mismatch, or attempt to be compatible.
-Bailing will result in a bad user experience without recourse, so we want to avoid that.
-Backwards compatibility is not straightforward though, and we want to limit its scope.
-While we only have one extension published, there are already several DApp implementations, so we will try to handle the compatibility in the extension: the Sporran.
-This will address the case when the extension speaks the newer version, and we will try to avoid the opposite case by releasing the upgraded Sporran much earlier.
-We will not handle the case when the user manually disabled Sporran updates.
+When upgrading to a 0.29.x version of the SDK and to the Credentials API version 3.0, we recommend backwards support of Credentials API version 2.0, as supporting only the latest version may result in poor user experience. In what follows, we outline an upgrade strategy for implementers of the Credentials API specification.
 
-Sporran should achieve compatibility by translating the messages sent to the previous versions of DApps.
+These instructions will also help with translating from and to data types of pre 0.29 SDK versions in other scenarios, such as when sending messages between clients, or when importing older data (e.g. credentials).
 
-For `request-attestation` credential as requestForAttestation (claimerSignature is not required by SKYC?).
+## General Strategy
+
+Since version 3.0, the specification requires conformant web apps as well as extensions to announce the versions of the API they use, allowing for version negotiation.
+Because extensions injects themselves into web pages that signal support for kilt features via the `window.kilt` property, the recommended strategy is to handle backwards compatibility on the extension side.
```

Comment:
```suggestion
Because extensions inject themselves into web pages that signal support for kilt features via the `window.kilt` property, the recommended strategy is to handle backwards compatibility on the extension side.
```

Reply:
whoops thanks